pkg/apis/k0s/v1beta1/clusterconfig_types.go

```diff
@@ -332,6 +336,64 @@ func (s *ClusterSpec) Validate() (errs []error) {
 		errs = append(errs, err)
 	}
 
+	for _, err := range s.validateKine(online) {
+		errs = append(errs, err)
+	}
+
+	return
+}
+
+func (s *ClusterSpec) validateKine(online bool) (errs field.ErrorList) {
+	if s.Storage.Type != KineStorageType {
+		return
+	}
+
+	if s.Storage.Kine == nil {
+		errs = append(errs, field.NotFound(field.NewPath("storage", "kine"), "missing"))
+		return
+	}
+
+	dataSource := s.Storage.Kine.DataSource
+
+	if dataSource == "" {
+		// sqlite - defaults
+		return
+	}
+
+	if !online {
+		return
+	}
+
+	path := field.NewPath("storage", "kine", "datasource")
+
+	u, err := url.Parse(dataSource)
+	if err != nil {
+		detail := "unable to parse datasource"
+		errs = append(errs, field.Invalid(path, dataSource, detail))
+		return
+	}
+
+	if u.Scheme == "sqlite" {
+		err = unix.Access(u.Path, unix.W_OK)
```

Comment:
Okay, Windows support is :wave: at us here. We need to factor this out into some _unix/_windows files, I fear, or find a solution that compiles on both Linux and Windows.

Reply:
K0s only supports Windows workers only, so the Windows implementation could simply panic, saying that it's unimplemented, I guess?